src/sentry/models/groupinbox.py

```diff
@@ -29,3 +29,20 @@ class GroupInbox(Model):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_groupinbox"
+
+
+def add_group_to_inbox(group, reason, reason_details=None):
+    if not GroupInbox.objects.filter(group=group).exists():
+        return GroupInbox.objects.create(
+            group=group, reason=reason.value, reason_details=reason_details
+        )
+    else:
+        return None
+
+
+def remove_group_from_inbox(group):
+    try:
+        group = GroupInbox.objects.get(group=group).delete()
+        group.delete()
```

Comment:
I think it might be worth testing these both directly as well as via `post_process`